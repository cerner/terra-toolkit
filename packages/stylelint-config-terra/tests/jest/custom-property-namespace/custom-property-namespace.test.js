/**
 * @jest-environment node
 */
const stylelint = require('stylelint');
const path = require('path');

const config = {
  plugins: [
    path.resolve(__dirname, '..', '..', '..', 'src', 'rules', 'custom-property-namespace', 'custom-property-namespace'),
  ],
  rules: {
    'terra/custom-property-namespace': true,
  },
};

const configCustomNamespace = {
  plugins: [
    path.resolve(__dirname, '..', '..', '..', 'src', 'rules', 'custom-property-namespace', 'custom-property-namespace'),
  ],
  rules: {
    'terra/custom-property-namespace': [
      true,
      {
        namespace: 'mock-namespace',
      }],
  },
};

describe('custom-property-namespace', () => {
  it('does not error with a default namespace', () => {
    const result = stylelint.lint({
      files: [
        path.resolve(__dirname, 'pass.css'),
      ],
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does error with no default namespace', () => {
    const result = stylelint.lint({
      files: [
        path.resolve(__dirname, 'fail.css'),
      ],
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });

  it('does not error with a custom namespace', () => {
    const result = stylelint.lint({
      code: '.test { color: var(--mock-namespace-color) }',
      config: configCustomNamespace,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does error with no custom namespace', () => {
    const result = stylelint.lint({
      code: '.test { color: var(--test-color) }',
      config: configCustomNamespace,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });
});
