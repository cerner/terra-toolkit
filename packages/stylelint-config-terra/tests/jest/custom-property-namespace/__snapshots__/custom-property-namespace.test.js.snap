// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-property-namespace does error with no custom namespace 1`] = `
Array [
  Object {
    "column": 9,
    "line": 1,
    "rule": "terra/custom-property-namespace",
    "severity": "error",
    "text": "Expected --test-color to be prefixed with the namespace: --mock-namespace (terra/custom-property-namespace)",
  },
]
`;

exports[`custom-property-namespace does error with no default namespace 1`] = `
Array [
  Object {
    "column": 9,
    "line": 1,
    "rule": "terra/custom-property-namespace",
    "severity": "error",
    "text": "Expected --test-color to be prefixed with the namespace: --stylelint-config-terra (terra/custom-property-namespace)",
  },
]
`;
